clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAO.scala

```diff
@@ -52,14 +53,15 @@ class HttpElasticsearchDAO private[dataaccess] (
   override def queryMetadata[D <: ClioDocument: HitReader](
     queryDefinition: QueryDefinition,
     index: ElasticsearchIndex[D]
-  ): Future[Seq[D]] = {
+  ): Source[D, NotUsed] = {
     val searchDefinition = search(index.indexName / index.indexType)
       .scroll(HttpElasticsearchDAO.DocumentScrollKeepAlive)
       .size(HttpElasticsearchDAO.DocumentScrollSize)
       .sortByFieldAsc(HttpElasticsearchDAO.DocumentScrollSort)
       .query(queryDefinition)
-    val searchResponse = httpClient execute searchDefinition
-    searchResponse flatMap foldScroll(Seq.empty[D], searchDefinition.keepAlive)
+
+    val responsePublisher = httpClient.publisher(searchDefinition)
+    Source.fromPublisher(responsePublisher).map(_.to[D])
```

Comment:
So this creates a lazy output stream?

Reply:
Yup! `httpClient.publisher` returns a Java-style reactive stream publisher, and we just plug it into Akka.